test/Microsoft.TemplateEngine.Utils.UnitTests/SemanticVersionTests.cs

```diff
@@ -0,0 +1,207 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Xunit;
+
+namespace Microsoft.TemplateEngine.Utils.UnitTests
+{
+    public class SemanticVersionTests
```

Comment:
Add a few tests that check:
* `bool Equals(object)`
* `int CompareTo(object)`
* `int GetHashCode()`
* `int CompareTo(SemanticVersion, out bool)` - this is being verified indirectly, but a test or two the checks the build metadata flag explicitly would be good
* `string ToString()`
* Tests verifying the behavior of null values with `bool TryParse(string, out SemanticVersion)` `int CompareTo(object)`/`int CompareTo(SemanticVersion)`, `int CompareTo(SemanticVersion, out bool)` and operators